runtime/hydradx/src/evm/executor.rs

```diff
@@ -1,25 +1,53 @@
-use evm::executor::stack::{StackExecutor, StackSubstateMetadata};
+use crate::evm::evm_fee::FeeCurrencyOverrideOrDefault;
+use crate::evm::runner::WrapRunner;
+use crate::evm::{EvmAccounts, WethAssetId};
+use crate::types::ShortOraclePrice;
+use crate::{DotAssetId, Runtime, XykPaymentAssetSupport};
+use core::marker::PhantomData;
 use evm::ExitFatal::Other;
-use evm::ExitReason;
-use fp_evm::Vicinity;
+use evm::{
+	executor::stack::{StackExecutor, StackState as StackStateT, StackSubstateMetadata},
+	Context as EvmContext, ExitError, ExitReason,
+};
+use fp_evm::{CallInfo, Vicinity};
 use frame_support::storage::with_transaction;
 use frame_support::traits::Get;
+use frame_system;
+use hydradx_adapters::price::ConvertBalance;
 use hydradx_traits::evm::{CallContext, EVM};
+use pallet_currencies::fungibles::FungibleCurrencies;
 use pallet_evm::runner::stack::SubstrateStackState;
+use pallet_evm::{
+	self, runner::Runner as EvmRunnerT, AccountProvider as EvmAccountProviderT, AddressMapping as EvmAddressMappingT,
+	Config as EvmConfigT, Pallet as EvmPallet,
+};
 use pallet_evm::{AccountProvider, AddressMapping, Config};
-use primitive_types::{H160, U256};
+use pallet_evm::{Error, Runner, RunnerError};
+use pallet_genesis_history::migration::Weight;
+use sp_core::{H160, H256, U256};
 use sp_runtime::{DispatchError, TransactionOutcome};
 use sp_std::vec;
 use sp_std::vec::Vec;
 
-pub struct Executor<R>(sp_std::marker::PhantomData<R>);
-
 pub type CallResult = (ExitReason, Vec<u8>);
 
 pub type BalanceOf<T> =
 	<<T as pallet_evm::Config>::Currency as frame_support::traits::Currency<pallet_evm::AccountIdOf<T>>>::Balance;
 pub type NonceIdOf<T> = <<T as Config>::AccountProvider as AccountProvider>::Nonce;
 
+pub struct Executor<R>(sp_std::marker::PhantomData<R>);
+
+type EVMRunner = WrapRunner<
+	Runtime,
+	pallet_evm::runner::stack::Runner<Runtime>, // Evm runner that we wrap
+	hydradx_adapters::price::FeeAssetBalanceInCurrency<
```

Comment:
do we really need it if we dont use it? cannot we just have some default impl there or do we even have to wrap it?

Reply:
or why dont we use runner we already wrapped for runtime?